specs/monitoring_integration_service.adoc

```diff
@@ -0,0 +1,166 @@
+= Spec - Monitoring-integration as a service
+
+== Introduction
+
+Monitoring Integration component is responsible for the integration of tendrl with grafana.
+It performs the following functionalities:
+
+* Uploads default dashboards to grafana.
+
+== Problem description
+
+In the present implementation of monitoring-integration to upload dashboards in
+grafana the user has to run the code manually every time which is not feasible.
+Monitoring integration will be providing other functionalities such as
+
+* Handling alerts generated by grafana
+* Making dynamic configuration changes in graphite and grafana
+
+for which it will be required to be implemented as a service.
+
+== Proposed change
+
+Monitoring-integration component is changed to a service based on the tendrl structure.
+Instead of starting each and every times manually, it is installed and started once
+as a service in the background.
+
+To convert as a service monitoring-integration component needs some additional
+modules on existing structure.
+
+1.Manager
+
+2.Namespace
+
+3.Object
+
+Mananger:
+
+Monitoring-integration execution flow is started from the manager modules. Manager
+module is the entry point for monitoring-integration, all the gevents are initialized
+and started from the manager modules. A manager can kill all the gevent when
+monitoring-integration service stops.
+
+When monitoring-integration start manager module will do the all configuration
+using configuration files. It will also create a separate namespace for the monitoring-integration.
+
+Namespace:
+
+A manager will create its own namespace for the monitoring-integration to separate
+its own objects from the other modules. These objects are only accessible by
+monitoring-integration.
+
+Object:
+
+Each tendrl component has its own set of objects for a different purpose. Each
+object have its own structure. Monitoring-integration have an object called Config.
+
+
+== Alternatives
+
+None
+
+== Data model impact:
+
+Monitoring-integration configuration stored in etcd : “_NS/monitoring_integration/config”
+
+== Impacted Modules:
+
+=== Tendrl API impact:
+
+None
+
+=== Notifications/Monitoring impact:
+
+None
+
+=== Tendrl/common impact:
+
+None
+
+=== Tendrl/node_agent impact:
+
+None
+
+=== Sds integration impact:
+
+None
+
+
+=== Tendrl/monitoring-integration impact:
+
+Changes need to convert monitoring-integration as service:
+
+* Create a module called manager in monitoring-integration
+* Create a namespace for the monitoring-integration
+* Create module called objects to place all monitoring-integration objects.
+
+The implementation of these three modules is in the implementation section.
+
+=== Security impact:
+
+None
+
+=== Other end user impact:
+
+None
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+Convert the monitoring-integration as service
+
+== Implementation:
```

Comment:
@shtripat In this spec, I just simply mentioned steps to change as a service. I have not mentioned anything about gevents. So I will cover that in another spec which is related to utilization aggregation calculation in monitoring integration.